### PR TITLE
[#151394645] Three unit tests for matches.

### PR DESCRIPTION
**Pull request by mpope9**
These are three unit tests for matches.  They test for connection, disconnection, and a legal amount of clients per match.  They are relatively simple, because they will need many changes in one or two stories.

to run:
```
premake5 test
./bin/test
```

Valgrind reported no extra errors or leaks.
to run:
```
valgrind ./bin/test
```

@kkaminski @cchung89 please review.

### CODE REVIEW

File: netcode.c [hunk 1]
```diff
@@ -7918,6 +7918,314 @@ void test_loopback()
...
+    struct netcode_server_t * server = netcode_server_create_internal("[::1]:40000", TEST_PROTOCOL_ID, private_key, time, network_simulator, NULL, NULL, NULL );
+    check( server );
+    netcode_server_start( server, num_clients );
+
+    struct netcode_client_t ** clients = (struct netcode_client_t **) malloc( sizeof( struct netcode_client_t* ) * num_clients );
+    check(clients);
+
+    uint64_t token_sequence = 0;
+
+    // Client things.
```
Comment (kkaminski) [inline netcode.c]: ?

File: netcode.c [hunk 2]
```diff
@@ -7918,6 +7918,314 @@ void test_loopback()
...
+
+    netcode_server_stop( server );
+    netcode_server_destroy( server );
+
+    netcode_network_simulator_destroy( network_simulator );
+
+    free( clients );
+}
+
+// This test will need massive changes when we introduce removing matches based on dc time.
```
Comment (kkaminski) [inline netcode.c]: Try to avoid acronyms - guessing you mean `disconnect`

File: netcode.c [hunk 3]
```diff
@@ -7918,6 +7918,314 @@ void test_loopback()
...
+    check( match->num_clients_in_match == num_clients );
+
+    for( int i = 0; i < num_clients; ++i )
+    {
+        netcode_server_disconnect_client( server, i );
+    }
+    for( int i = 0; i < num_clients; ++i )
+    {
+        netcode_client_destroy( clients[i] );
+    }
```
Comment (kkaminski) [inline netcode.c]: do these need to be 2 separate loops?  IE do you need to disconnect both clients before destroying them?

File: netcode.c (hunk 1 above)
Comment (mpope9) [reply netcode.c]: Are you asking about the token sequence?  Or the vague comment?

File: netcode.c (hunk 3 above)
Comment (mpope9) [reply netcode.c]: I'll separate them.

File: netcode.c (hunk 2 above)
Comment (mpope9) [reply netcode.c]: Will change.

File: netcode.c (hunk 1 above)
Comment (mpope9) [reply netcode.c]: Or the silly array access?

File: netcode.c (hunk 3 above)
Comment (mpope9) [reply netcode.c]: I mean combine

File: netcode.c (hunk 1 above)
Comment (kkaminski) [reply netcode.c]: sorry, the vague comment ;-)

**Comment by cchung89** [comment]
📦 LGTM

File: netcode.c (hunk 1 above)
Comment (kkaminski) [reply netcode.c]: Bump just bc I see you fixed this in one test but not here?

**Comment by kkaminski** [comment]
Just the bump for the other vague comment, otherwise 🚅

**Pull request merged**